pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -227,13 +228,13 @@ func (r *DNSPolicyReconciler) deleteResources(ctx context.Context, dnsPolicy *v1
 		return err
 	}
 
-	// clean up condition on gateway
-	if err := r.updateGatewayCondition(ctx, metav1.Condition{Type: string(DNSPolicyAffected)}, gatewayDiffObj); err != nil {
+	// update annotation of policies affecting the gateway
+	if err := r.ReconcileGatewayPolicyReferences(ctx, dnsPolicy, gatewayDiffObj); err != nil {
 		return err
 	}
 
-	// update annotation of policies affecting the gateway
-	return r.ReconcileGatewayPolicyReferences(ctx, dnsPolicy, gatewayDiffObj)
+	// remove gateway policy affected condition status
+	return r.updateGatewayCondition(ctx, metav1.Condition{Type: string(DNSPolicyAffected)}, gatewayDiffObj)
```

Comment:
I changed the order of the requests here as updating the status last makes more sense to me, and seems logical that it's the last thing removed in the case of deletion.